File: ympa_fsharp_giraffe_server/src/model/OrdersStatsPriceDTO.fs
namespace ympa_fsharp_giraffe_server.Model

open System
open System.Collections.Generic
open ympa_fsharp_giraffe_server.Model.OrdersStatsPriceType

module OrdersStatsPriceDTO =

  //#region OrdersStatsPriceDTO


  type OrdersStatsPriceDTO = {
    Type : OrdersStatsPriceType;
    CostPerItem : decimal;
    Total : decimal;
  }
  //#endregion

File: app/model/OrdersStatsPriceDTO.scala
package model

import play.api.libs.json._

/**
  * Цена или скидки на товар.
  * @param costPerItem Цена или скидка на единицу товара в заказе. Указана в рублях. Точность — два знака после запятой. 
  * @param total Суммарная цена или скидка на все единицы товара в заказе. Указана в рублях. Точность — два знака после запятой. 
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2024-09-09T22:33:09.055496365Z[Etc/UTC]", comments = "Generator version: 7.8.0")
case class OrdersStatsPriceDTO(
  `type`: Option[OrdersStatsPriceType],
  costPerItem: Option[BigDecimal],
  total: Option[BigDecimal]
)

object OrdersStatsPriceDTO {
  implicit lazy val ordersStatsPriceDTOJsonFormat: Format[OrdersStatsPriceDTO] = Json.format[OrdersStatsPriceDTO]
}

